src/lib/mina_networking/rpcs.ml

```diff
@@ -412,17 +453,20 @@ module Answer_sync_ledger_query = struct
     let ledger_hash, _ = Envelope.Incoming.data request in
     let query = Envelope.Incoming.map request ~f:Tuple2.get2 in
     let%bind answer =
-      let%bind.Deferred.Option frontier = return (get_transition_frontier ()) in
-      Sync_handler.answer_query ~frontier ledger_hash query ~logger
-        ~trust_system
+      match get_transition_frontier () with
+      | Some frontier ->
+          Sync_handler.answer_query ~frontier ledger_hash query
+            ~context:(module Context)
+            ~trust_system
+      | None ->
+          return (Or_error.error_string "No Frontier")
     in
     let result =
-      Result.of_option answer
-        ~error:
-          (Error.createf
-             !"Refusing to answer sync ledger query for ledger_hash: \
-               %{sexp:Ledger_hash.t}"
-             ledger_hash )
+      Result.map_error answer ~f:(fun e ->
+          Error.createf
+            !"Refusing to answer sync ledger query for ledger_hash: \
+              %{sexp:Ledger_hash.t}. Error: %s"
+            ledger_hash (Error.to_string_hum e) )
```

Comment:
This is confusing to me: why are we providing an error message `No frontier` in `answer` to just immediately replace it with `map_error`?

Relatedly, we then match on `result` below, so we will record `Trust_system.Actions.Requested_unknown_item` even if it's our fault for not having a frontier.

Reply:
This behaviour is unchanged from before, so I won't block on it.

Reply:
Yeah, I'm only propagating the actual error to test error cases. 
> we will record Trust_system.Actions.Requested_unknown_item even if it's our fault for not having a frontier.

This seems to be the behaviour for other messages too. Opened a ticket #16383